src/torchjd/autojac/_transform/_differentiate.py

```diff
@@ -29,7 +29,13 @@ def _compute(self, tensors: _A) -> _A:
 
     @abstractmethod
     def _differentiate(self, tensor_outputs: Sequence[Tensor]) -> tuple[Tensor, ...]:
-        raise NotImplementedError
+        """
+        Abstract method for differentiating the outputs with respect to the inputs, and
+        right-multiplying them with the provided tensor_outputs.
+
+        The implementation of this method should define what kind of differentiation is performed
+        (whether gradients, Jacobians, etc. are computed).
```

Comment:
In no other of the docstrings for abstract method you said "Abstract method for [...]", I think we can avoid it here too. Otherwise we could add it everywhere, but I think it would be good to uniformize.

Reply:
I would see something along the lines of (any mix of your version and mine is fine, including none of my version)
```
Differentiate the outputs with respect to the inputs, and right-multiply the result with the provided `tensor_outputs`.

The implementations of this method specify the type of differentiation performed (gradients scaled with a scalar, Jacobians multiplied by a vector, etc.).
```

Reply:
In all abstract methods except this one, the role of the method is clear from its name and maybe a simple docstring. Here, it's not clear that "Differentiate" generalizes the concept of "computing gradients" and "computing Jacobians". So this is the only case where we need to be extra precise in the base docstring and override the docstring in the overridden methods.

This implies that this docstring has to be overridden by all non-abstract methods, so it's fine to talk about abstract things here in my opinion.

But yeah I agree that we could have a better wording, I'll look into it.